expression/builtin_time.go

```diff
@@ -1852,7 +1852,66 @@ type builtinPeriodDiffSig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_period-diff
 func (b *builtinPeriodDiffSig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("PERIOD_DIFF")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+	if args[0].IsNull() || args[1].IsNull() {
+		return d, nil
+	}
+	// convert input to int64
+	sc := b.ctx.GetSessionVars().StmtCtx
+	period1, err := args[0].ToInt64(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	period2, err := args[1].ToInt64(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	//input check format
+	if period1 <= 0 || period2 <= 0 || period1 > 999999 || period2 > 999999 {
+		d.SetNull()
```

Comment:
d is still null now.
It is better to check period1 immediately after you get period1.